src/backend/metal/src/native.rs

```diff
@@ -294,6 +295,7 @@ pub enum ImageLike {
         descriptor: metal::TextureDescriptor,
         mip_sizes: Vec<buffer::Offset>,
         host_visible: bool,
+        name: RefCell<String>,
```

Comment:
How can we use `RefCell` here if it's not meant to be used by multiple threads? I think we have to use `Mutex` instead.